### PR TITLE
Add App to Orchestrator and allow Name to be read from it

### PR DESCRIPTION
**Pull request by patrick-strata**

### CODE REVIEW

**Comment by patrick-strata** [comment]
This PR needs to be merged and a new service-extension library release be created.
After that the new lib should be referenced in the corresponding orchestrator PR:

**Review by daniilstrata — APPROVED** [review]
LGTM! Just one nit

File: app/app.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package app
+
+type App interface {
+	Name() string
```
Comment (daniilstrata) [inline app/app.go:4]: Please, add godoc.

Comment (patrick-strata) [reply app/app.go:7]: Thanks for catching this @daniilstrata, I pushed a commit to address it.

**Pull request merged**